src/packages/utils/src/things/json-rpc/json-rpc-quantity.ts

```diff
@@ -131,19 +134,100 @@ export class Quantity extends BaseJsonRpcType {
     return firstNonZeroByte;
   }
 
-  static toBuffer(value: JsonRpcInputArg, nullable?: boolean): Buffer {
+  private static getArgumentAsBigInt(argument: JsonRpcInputArg | Quantity): bigint {
+    const argumentType = typeof argument;
+    let argumentValue: bigint;
+
+    if (argumentType === "number") {
+      const argumentNumber = argument as number;
+      if (argumentNumber % 1) {
+        throw new Error("Cannot wrap a decimal as a json-rpc type.");
+      }
+      if (!isFinite(argumentNumber)) {
+        throw new Error(`Cannot wrap ${argument} as a json-rpc type.`);
+      }
+      argumentValue = BigInt(argumentNumber);
+    }  else if (argumentType === "bigint") {
+      argumentValue = argument as bigint;
+    } else if (argumentType === "string") {
+      if ((argument as string).slice(0, 2).toLowerCase() !== "0x") {
+        throw new Error(`Cannot wrap "${argument}" as a JSON-RPC type; strings must be hex-encoded and prefixed with "0x".`);
+      }
+      argumentValue = BigInt(argument as string);
+    } else if (argument instanceof Quantity) {
+      argumentValue = argument.toBigInt();
+    } else if (Buffer.isBuffer(argument)) {
+      argumentValue = bufferToBigInt(argument);
+    } else if (argumentValue == null) {
+      argumentValue = 0n;
+    }
+
+    return argumentValue;
+  }
+
+  public add(addend: JsonRpcInputArg | Quantity): Quantity {
+    if (this.bufferValue == null || this.bufferValue.length === 0) {
+      return addend instanceof Quantity ? addend : Quantity.from(addend, this._nullable);
```

Comment:
This check will fail if the `Quantity` object passed in is not an instance of this `Quantity` class. We should probably check for `"toBigInt" in argument` or something like that.